polyfill/lib/datetime.mjs

```diff
@@ -445,18 +445,18 @@ export class DateTime {
     const roundingIncrement = ES.ToTemporalRoundingIncrement(options, maximumIncrements[smallestUnit], false);
 
     let { deltaDays, hours, minutes, seconds, milliseconds, microseconds, nanoseconds } = ES.DifferenceTime(
-      GetSlot(this, HOUR),
```

Comment:
`this` and `other` are swapped here. That is probably where the test failures are coming from.